### ISSUE TITLE
jsdom does not update selectionStart when `value` of `<input type=text>` is set

### ISSUE THREAD

**Issue by Matthew-Goldberg** [jsdom/jsdom#2787]
### Basic info:

- **Node.js version:** 10.15.3
- **jsdom version:** 15.2.1

jsdom does not update selectionStart when `value` of `<input type=text>` is set.

Actual: selectionStart is 0
Expected: selectionStart should be at the end of the set text

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

const dom = new JSDOM(`<input type="text" value="4444" />`);
const { document } = dom.window;
const input = document.querySelector("input");
console.log(`Before: selectionStart=${input.selectionStart}, Value=${input.value}`);
// Set the value
input.value = 55555;
console.log(`After: selectionStart=${input.selectionStart}, Value=${input.value}`);
```

Result:
```
Before: selectionStart=0, Value=4444
After: selectionStart=0, Value=55555
```

### How does similar code behave in browsers?
In browser, this behaves differently.  The selectionStart after setting the value to `55555` is `5`, not 0.

See codepen: https://codepen.io/matthewg0/pen/YzPLyye

Result:
```
Before: selectionStart=0, Value=4444
After: selectionStart=5, Value=55555
```

You can also see this expected behavior in headless browser, using the following code:
```js
const puppeteer = require("puppeteer");

(async () => {
  // Now test in browser
  const htmlContent = `<input type="text" value="4444">`;
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.setContent(htmlContent);
  const content = await page.content();

  console.log("--- Chrome ---");
  console.log(content);
  let input = await page.$eval("input", el => ({ selectionStart: el.selectionStart, value: el.value }));
  console.log(`Before: selectionStart=${input.selectionStart}, Value=${input.value}`);
  await page.$eval("input", el => (el.value = 55555));
  input = await page.$eval("input", el => ({ selectionStart: el.selectionStart, value: el.value }));
  console.log(`After: selectionStart=${input.selectionStart}, Value=${input.value}`);
  await browser.close();
})();
```

Result:
```
--- Chrome ---
<html><head></head><body><input type="text" value="4444"></body></html>
Before: selectionStart=0, Value=4444
After: selectionStart=5, Value=55555
```

**Comment by domenic** [comment]
Great bug report!

My guess is that this was implemented according to the spec at the time, but we fixed the spec to match browser behavior in https://github.com/whatwg/html/pull/3472. Re-implementing the selection-tracking logic in jsdom to match the newest spec, and thus pass the tests at https://wpt.fyi/results/html/semantics/forms/textfieldselection?label=experimental&label=master&aligned, would be much appreciated.

/cc @TimothyGu who has been doing a lot of form-control-related work recently.
